rpms/awips2.edex/Installer.edex/programs/updateNDM.pl

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/perl
+
+#This script gets installed with EDEX and a cronjob is setup to run this script daily. This script initially sets up a local git repo and is configured to just checkout the National Dataset files from Unidata's public git repo and then copies the files to /awips2/edex/data/ndm to keep the databases updated with new metar/station information.
+
+$ndmGitPath = "rpms/awips2.edex/Installer.edex/ndm/";
+
+#get the current branch/version of awips2
+$version = `grep AWIPSII_VERSION /awips2/repo/awips2/build/buildEnvironment.sh`;
+chomp $version;
+@split = split(/"/,$version);
+
+$branch = "unidata_$split[1]";
+
+#Check if ndm location has been created
+$ndmDir="/awips2/dev/ndm";
+if(!-d $ndmDir)
+{
+  `mkdir $ndmDir`;
+}
+
+#Check if ndm git files have been checked out yet
+if(!-e "$ndmDir/.git")
+{
+  `cd $ndmDir ; 
+   git init ; 
+   git remote add -f origin https://github.com/Unidata/awips2.git ; 
+   git config core.sparseCheckout true ;
+   echo \"$ndmGitPath\" >> .git/info/sparse-checkout ;
+   git checkout -t origin/$branch
+  `;
+}
+
+print "\n\n\n";
+
+#Pull latest files
+`cd $ndmDir ; 
+ git pull origin $branch
+`;
+
+#copy files to AWIPS-II NDM endpoint
+`rsync -aP $ndmDir/$ndmGitPath /awips2/edex/data/ndm/`;
```

Comment:
this is recursive (will do all files in the directory)?  So, there will be a subdirectory in `/awips2/edex/data/ndm/` called `/rpms/awips2.edex/Installer.edex/ndm/`?  That seems excessive, unless this is doing something different from what I'm thinking?
And you're using `rsync` instead of `cp` because it's smarter and won't copy files that haven't changed?

Reply:
Also, why not do this command in the previous \` \` block?

Reply:
no, it will just copy the files in `/rpms/awips2.edex/Installer.edex/ndm/` directly to `/awips2/edex/data/ndm/`. Nothing about the initial path is kept (there are flags that do that, but this does not). 

And yea, I almost always use `rsync` because it's faster and smarter than `cp`.

Reply:
Definitely could be done in the previous code block